docs/setup.asciidoc

```diff
@@ -34,19 +41,104 @@ https://www.nuget.org/packages/Elastic.Apm.SqlClient[**Elastic.Apm.SqlClient**]:
+
+
+[source,csharp]
+----
+using Elastic.Apm.NetCoreAll;
+
+namespace MyApplication
+{
+  public class Program
+  {
+    public static IHostBuilder CreateHostBuilder(string[] args) =>
+        Host.CreateDefaultBuilder(args)
+            .ConfigureWebHostDefaults(webBuilder => { webBuilder.UseStartup<Startup>(); })
+            .UseAllElasticApm();
+
+    public static void Main(string[] args) => CreateHostBuilder(args).Build().Run();
+  }
+}
+----
+
+With the `UseAllElasticApm()` the agent with all its components is turned on. On ASP.NET Core it'll automatically capture incoming requests, database calls through supported technologies, outgoing HTTP requests and so on.
+
+Additionally the `UseAllElasticApm` will also add an `ITracer` to the Dependency Injection system. This means you can use the <<public-api>> in your code like this:
+
+[source,csharp]
+----
+using Elastic.Apm.Api;
+
+namespace WebApplication.Controllers
+{
+    public class HomeController : Controller
+    {
+        private readonly ITracer _tracer;
+        public HomeController(ITracer tracer) //inject ITracer
+        => _tracer = tracer;
+
+        public IActionResult Index()
+        {
+            //use ITracer
+            var span = _tracer.CurrentTransaction?.StartSpan("MySampleSpan", "Sample");
+            try
+            {
+                //your code here
+            }
+            catch (Exception e)
+            {
+                span?.CaptureException(e);
+                throw;
+            }
+            finally { }
+            {
+                span?.End();
+            }
+            return View();
+        }
+    }
+}
+----
+
+Similarly to this ASP.NET Core controller, you can use the same approach with `IHostedService` implementations.
+
+The `Elastic.Apm.NetCoreAll` will reference every agent components. This is usually not a problem. Nevertheless in case you want to keep dependencies minimal you can also reference the `Elastic.Apm.Extensions.Hosting` and use the `UseElasticApm` method instead of `UseAllElasticApm`. With this you can control what the agent will listen for.
+
+For example the following only turns on outgoing HTTP monitoring (so, for example database or elasticsearch calls won't be automatically captured):
+
+[source,csharp]
+----
+    public static IHostBuilder CreateHostBuilder(string[] args) =>
+        Host.CreateDefaultBuilder(args)
+            .ConfigureWebHostDefaults(webBuilder => { webBuilder.UseStartup<Startup>(); })
+            .UseElasticApm(new HttpDiagnosticsSubscriber());
+
+----
+
+
 [float]
 [[setup-asp-net-core]]
 ==== ASP.NET Core
 
+We suggest using the approach described in the <<setup-dotnet-net-core, .NET Core setup instructions>> section above. We keep the `IApplicationBuilder` introduced here only for backwards compatibility.
```

Comment:
With the approach documented above (the `IHostBuilder` thing) we basically made things that are discussed here unnecessary. Long term I think we just need to deprecate this `ASP.NET Core` section.